### PR TITLE
[libigl] Always install as header-only library

### PR DESCRIPTION
**Pull request by FabienPean**
In #14376, triplet *-windows-dynamic version was updated to rely on the library as header-only while leaving the *-static triplet intact. However, the static library archive still suffers from the same reported issues: huge size and not all combination of explicitly instantiated templates exist. This PR forces installs of libigl as header-only library for both static and dynamic triplets. (Particularly useful for vcpkg on linux where the default triplet has static linkage by default)

Let me know if this PR is an agreeable change on your side.

### CODE REVIEW

**Comment by FabienPean** [comment]
The PR shows failure for another port.

File: ports/libigl/install-extra-headers.patch [hunk 1]
```diff
@@ -2,19 +2,17 @@ diff --git a/cmake/libigl.cmake b/cmake/libigl.cmake
...
 +++ b/cmake/libigl.cmake
-@@ -560,6 +560,85 @@ export(
+@@ -560,6 +560,75 @@ export(
  # Install headers for core library
  install_dir_files(core)
  install_dir_files(copyleft)
 + 
 +if (LIBIGL_WITH_EMBREE)
 +    install_dir_files(embree)
-+#    if(NOT LIBIGL_USE_STATIC_LIBRARY)
```
Comment (JackBoosY) [inline ports/libigl/install-extra-headers.patch:12]: This issue seems like a upsteam issue, can you report this to the upstream?
And does the upstream accept this changes?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Maybe we should add an new feature `header-only` to this port instead of force it to be a header only port.

**Comment by FabienPean** [comment]
Added header-only as a feature

File: ports/libigl/portfile.cmake [hunk 2]
```diff
@@ -14,7 +14,7 @@ vcpkg_from_github(
 )
 
 set(LIBIGL_BUILD_STATIC OFF)
-if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
+if (NOT "header-only" IN_LIST FEATURES AND VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (JackBoosY) [inline ports/libigl/portfile.cmake:17]: Should this judgment be `("header-only" IN_LIST FEATURES) OR (VCPKG_LIBRARY_LINKAGE STREQUAL static)` ?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libigl/portfile.cmake (hunk 2 above)
Comment (FabienPean) [reply ports/libigl/portfile.cmake:17]: The full block is
> set(LIBIGL_BUILD_STATIC OFF)
if (NOT "header-only" IN_LIST FEATURES AND VCPKG_LIBRARY_LINKAGE STREQUAL static)
    set(LIBIGL_BUILD_STATIC ON)
endif()

So it sounds correct to me like that. LIBIGL is built in static mode only when there is static linkage and header-only is *not* demanded by the user.

Comment (JackBoosY) [reply ports/libigl/portfile.cmake:17]: Fine.

**Review by JackBoosY — APPROVED** [review]

**Review by BillyONeal — CHANGES_REQUESTED** [review]
The "header-only" feature doesn't comply with our guidelines for features, namely, that they do not provide "alternative" selections. See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-use-features-to-control-alternatives-in-published-interfaces

Crucially, any downstream consumer who is passing -ligl (or similar to link the static lib) would be broken by turning on this "feature".

It might be acceptable to make such a chance unconditional if the current users of the port think that makes sense.

/cc @patrikhuber @deGravity @xarthurx

**Review by ras0219-msft — APPROVED** [review]
Since the comments in the upstream project indicate that it is a "header only library", I think the right way forward is to always just do that.

With this change, LGTM!

**Pull request merged**